SmartDeviceLink/SDLUploadFileOperation.m

```diff
@@ -64,99 +69,165 @@ - (void)sdl_sendPutFiles:(NSArray<SDLPutFile *> *)putFiles withCompletion:(SDLFi
     dispatch_group_t putFileGroup = dispatch_group_create();
     dispatch_group_enter(putFileGroup);
 
-    // When the putfiles all complete, run this block
+    // Waits for all packets be sent before returning whether or not the upload was a success
     __weak typeof(self) weakself = self;
     dispatch_group_notify(putFileGroup, dispatch_get_main_queue(), ^{
         if (streamError != nil || stop) {
             completion(NO, bytesAvailable, streamError);
         } else {
             completion(YES, bytesAvailable, nil);
         }
-
         [weakself finishOperation];
     });
 
-    for (SDLPutFile *putFile in putFiles) {
+    NSInputStream *inputStream = [self sdl_openInputStreamWithFile:file];
+
+    // Break the data into small pieces, each of which will be sent in a separate putfile
+    unsigned long long fileSize = [file fileSize];
```

Comment:
No need for this temp variable. In the two places its used, replace with `file.fileSize`